pallets/pool-system/src/tranches.rs

```diff
@@ -1323,7 +1326,7 @@ where
 						.checked_pow(
 							n_tranches
 								.checked_sub(tranche.seniority)
-								.unwrap_or(u32::MAX),
+								.unwrap_or(u32::MAX), //u32::MAX sure?
```

Comment:
Seems like if the subtraction fails, it should be 0 here. Maybe it should be a `saturating_sub`?

Reply:
I am actually not sure. The create logic ensures we have always `tranche.seniority <= tranche.len()` and `tranche.seniority` greater/equal to seniority of previous tranche.

Reply:
I agree that if this is not possible, we should return an error or ensure the expect will never fail (even if in the future some precondition in other places changes)

Reply:
Agreed, if we have a specific test for this.